gix/src/repository/config/transport.rs

```diff
@@ -270,7 +274,7 @@ impl crate::Repository {
                         .proxy
                         .as_deref()
                         .filter(|url| !url.is_empty())
-                        .map(|url| gix_url::parse(url.into()))
+                        .map(|url| gix_url::parse(<&BStr>::from(url)))
```

Comment:
It's interesting that this needed to change as it seems unrelated to the signature of the method affected by `momo`.

Reply:
Yeah it's really a shame.
momo currently works by "search and replace", which is why this is a problem.

A better approach would be to build a syntax tree and understand which variable it's referring to, which is quite complex and not sure whether I can do it within one PR.

Even better would be to add this to rustc.

Reply:
Despite valuing the effort you would be willing to make, I think it can be better spent - making some adjustments by hand seems like a small price to pay to not have to write a lot of boilerplate.